cinema/views.py

```diff
@@ -1 +1,67 @@
-# write views here
+from rest_framework import viewsets
+
+from cinema.models import Movie, MovieSession, Genre, CinemaHall, Actor
+from cinema.serializers import (
+    MovieSerializer,
+    MovieListSerializer,
+    MovieDetailSerializer,
+    MovieSessionSerializer,
+    MovieSessionListSerializer,
+    MovieSessionDetailSerializer,
+    GenreSerializer,
+    CinemaHallSerializer,
+    ActorSerializer
+)
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class CinemaHallViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.prefetch_related("genres", "actors")
+    serializer_class = MovieSerializer
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+
+        if self.action == "retrieve":
+            return MovieDetailSerializer
+
+        return MovieSerializer
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = MovieSession.objects.select_related("cinema_hall", "movie")
+    serializer_class = MovieSessionSerializer
+
+    def get_queryset(self):
+        if self.action == "list":
+            return self.queryset
+        if self.action == "retrieve":
+            return self.queryset.prefetch_related(
```

Comment:
In this case it doesn't work, you also have 7 queries in your DB, you should use `select_related()` here and you will have 5 queries in your DB

Reply:
I have static numb of queries with prefetch_related. I use info about many actors and many genres. So, select_related is bad for this situation. It's my opinion...
![image](https://user-images.githubusercontent.com/72568844/202929249-2ed4f286-d021-4b9f-8bd4-8b30b49d606a.png)

Reply:
Ok, I got you. I think the mentor will give us the answer, which is better to use in this case)

Reply:
@ketstap162 I did not quite understand where you got your statistics. Tell me in more detail why you decided that way?

Reply:
There are no statistics, I just learned that prefetch_related is used for ManyToMany relation. I discussed this with colleagues, the django interface provides this under the hood (if I'm not mistaken), but I would like to hear the exact version from you